src/odemis/driver/orsay.py

```diff
@@ -1389,3 +1440,1247 @@ def terminate(self):
+        self.connect(va, parameter, attributeName)
+
+    def connect(self, va, parameter, attributeName="Actual"):
+        """
+        Subscribes the VA to the parameter
+
+        :param (VigilantAttribute) va: The vigilant attribute this Orsay parameter connector should be connected to.
+            This VA should not have a setter yet, because the setter will be overwritten.
+        :param (Orsay Parameter) parameter: A parameter of the Orsay server. It can also be a list of parameters, if va
+            can contain a Tuple of equal length.
+        :param (string) attributeName: The name of the attribute of parameter the va should be synchronised with.
+            Defaults to "Actual".
+        """
+        # Log a warning when using the connector to connect a parameter to a VA, whilst the connector is already in use
+        if self._parameters is not None and None not in {self._attributeName, self._va, self._va_type_name}:
+            logging.warning("OrsayParameterConnector is already connected to an Orsay parameter. It is better to call "
+                            "disconnect before reconnecting to something else.")
+
+        # Assure that self._parameters (and self._minpar and self._maxpar if applicable) is a list
+        if type(parameter) in {set, list, tuple}:  # if multiple parameters are passed
+            self._parameters = list(parameter)
+            if self._minpar is not None and self._maxpar is not None:
+                self._minpar = list(self._minpar)
+                self._maxpar = list(self._maxpar)
+        else:  # if just one parameter is passed
+            self._parameters = [parameter]
+            if self._minpar is not None and self._maxpar is not None:
+                self._minpar = [self._minpar]
+                self._maxpar = [self._maxpar]
+
+        # Check that the number of parameters passed make sense
+        if len(self._parameters) == 0:
+            raise ValueError("No parameters passed")
+        if self._minpar is not None and self._maxpar is not None and (len(self._parameters) != len(self._minpar) or
+                                                                      len(self._parameters) != len(self._maxpar)):
+            raise ValueError("Number of parameters, minimum parameters and maximum parameters is not equal")
+
+        # Store and analyse the passed VA, to determine its type, if it's a tuple or not and if it's read-only
+        self._attributeName = attributeName
+        self._va = va
+        self._va_type_name = va.__class__.__name__
+        if self._va_type_name.startswith("Tuple"):
+            self._va_is_tuple = True
+            self._va_value_type = type(self._va.value[0])
+        else:
+            self._va_is_tuple = False
+            self._va_value_type = type(self._va.value)
+        if not self._va.readonly:  # only overwrite the VA's setter if the VA is not read-only
+            self._va._setter = WeakMethod(self._update_parameter)
```

Comment:
Subscribing actually cannot work here, because that causes callback loops.
Since _update_va is subscribed to the parameter, the va will change value when the parameter changes value. If then we subscribe _update_parameter to the va, then the va value will again be written to the parameter and so on. So here we need some way that the parameter will be updated if the va changes value, except for when the va changes value because the parameter changed value. That is now done through the setter. There are of course other ways, but I can't think of a nice one.

Reply:
I understand now. In other cases, to avoid such loop, we unsubscribe, set the value, (re)subscribe. However, it's quite ugly, and admittedly here, the use of setter is more correct. We could try to have a setter for the setter in the VA class, but let's not do that now... so leave this code as-is.